### PR TITLE
Add Matplotlib geo projections

### PR DESCRIPTION
**Pull request by QuLogic**
## Rationale

In Matplotlib, there are 4 builtin geo projections: Aitoff, Hammer, Mollweide, and Lambert. They are not quite as flexible as Cartopy, but work for maps at least. We have the latter two, but not the former.

## Implications

This adds Aitoff and Hammer. In proj, [Hammer support two more parameters](https://proj4.org/operations/projections/hammer.html): `+W` and `+M`. I do not know what these stand for, so I have not added them, but would like to if anyone has any understanding of what they mean.

### CODE REVIEW

**Comment by QuLogic** [comment]
Any thoughts on Hammer parameters?

It seems like 4.9.3 might be buggy with one of these.

**Comment by pelson** [comment]
> Any thoughts on Hammer parameters?

As in ``+W`` and ``+M``? I'm quite content to not provide these until they are needed (in the form of a PR from the person that does). 

The test is failing with a lot of warnings, but they may be not an issue? The image that now has all of the projections is (copied from Travis):

full

**Comment by greglucas** [comment]
Interest in coming back to these projections?

**Comment by QuLogic** [comment]
Rebased, but I did not add the other parameters. I'm also not sure if the threshold needs setting nowadays.

File: lib/cartopy/crs.py [hunk 1]
```diff
@@ -2112,6 +2112,45 @@ def y_limits(self):
...
+        globe: :class:`cartopy.crs.Globe`, optional
+            If omitted, a default globe is created.
+
+            .. note::
+                This projection does not handle elliptical globes.
+
+        """
+        proj_params = [('proj', 'aitoff'),
+                       ('lon_0', central_longitude)]
+        super(Aitoff, self).__init__(proj_params, central_longitude,
```
Comment (greglucas) [inline lib/cartopy/crs.py:2147]:
```suggestion
        super().__init__(proj_params, central_longitude,
```
You must have had this sitting around for a while needing to rebase from Python 2 supers :)

File: lib/cartopy/tests/crs/test_aitoff.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# (C) British Crown Copyright 2021, Met Office
```
Comment (greglucas) [inline lib/cartopy/tests/crs/test_aitoff.py:1]: Should change this to the new license file header.

File: lib/cartopy/crs.py (hunk 1 above)
Comment (QuLogic) [reply lib/cartopy/crs.py:2147]: Yea, I did catch the `__future__` imports at least.

**Comment by QuLogic** [comment]
Oops, created the test images with the oldest supported Matplotlib instead of the latest.

File: lib/cartopy/crs.py [hunk 3]
```diff
@@ -2117,6 +2117,45 @@ def y_limits(self):
...
+            The central longitude. Defaults to 0.
+        false_easting: float, optional
+            X offset from planar origin in metres. Defaults to 0.
+        false_northing: float, optional
+            Y offset from planar origin in metres. Defaults to 0.
+        globe: :class:`cartopy.crs.Globe`, optional
+            If omitted, a default globe is created.
+
+            .. note::
+                This projection does not handle elliptical globes.
```
Comment (greglucas) [inline lib/cartopy/crs.py:2147]: It doesn't look like this note appears anywhere in the docs? It seems like all of the __init__ docstrings are left out currently.

**Review by greglucas — APPROVED** [review]
I just had one minor comment, which should probably be a follow-up PR. Looks good to me otherwise.

File: lib/cartopy/crs.py (hunk 3 above)
Comment (QuLogic) [reply lib/cartopy/crs.py:2147]: Hmm, maybe it was with the change in doc themes? I don't remember what the older one looked like now.

**Comment by greglucas** [comment]
Sorry, I actually meant to merge this in before the release and just forgot.

File: lib/cartopy/crs.py (hunk 3 above)
Comment (greglucas) [reply lib/cartopy/crs.py:2147]: We can open up another issue if we want to update the docs later.

**Pull request merged**